src/core/map/coord.rs

```diff
@@ -474,28 +474,31 @@ mod coord_tests {
 	}
 
 	#[test]
-	fn coord_orientation_ring_iterator_count() {
+	fn coord_orientation_ring_iterator_small_count() {
 		{
 			let mut iter = CoordOrientationRingIterator::new(0);
 			assert_eq!(iter.next(), Some(CoordOrientation::new_axial(0, 0)));
 			assert_eq!(iter.next(), None);
 		}
 		{
-			let mut iter = CoordOrientationRingIterator::new(1);
-			assert_eq!(iter.next(), Some(CoordOrientation::new_axial(1, 0)));
-			assert_eq!(iter.next(), Some(CoordOrientation::new_axial(0, 1)));
-			assert_eq!(iter.next(), Some(CoordOrientation::new_axial(-1, 1)));
-			assert_eq!(iter.next(), Some(CoordOrientation::new_axial(-1, 0)));
-			assert_eq!(iter.next(), Some(CoordOrientation::new_axial(0, -1)));
-			assert_eq!(iter.next(), Some(CoordOrientation::new_axial(1, -1)));
-			assert_eq!(iter.next(), None);
+			let coords =
+				CoordOrientationRingIterator::new(1).collect::<HashSet<CoordOrientation>>();
+			assert!(coords.contains(&CoordOrientation::new_axial(1, 0)));
+			assert!(coords.contains(&CoordOrientation::new_axial(0, 1)));
+			assert!(coords.contains(&CoordOrientation::new_axial(-1, 1)));
+			assert!(coords.contains(&CoordOrientation::new_axial(-1, 0)));
+			assert!(coords.contains(&CoordOrientation::new_axial(0, -1)));
+			assert!(coords.contains(&CoordOrientation::new_axial(1, -1)));
+			assert_eq!(coords.len(), 6);
 		}
-		for distance in 1..5u8 {
-			let iter = CoordOrientationRingIterator::new(distance);
-			let mut around = HashSet::<CoordOrientation>::new();
-			for c in iter {
-				around.insert(c);
-			}
+	}
+
+	proptest!(
+		#![proptest_config(ProptestConfig::with_cases(30))]
+		#[test]
+		fn coord_orientation_ring_iterator_big_count(distance in 2..128u8) {
```

Comment:
Oh hey, didn't know proptest supported ranges.  You can have it test starting at `0` if you want, I just wasn't because I was testing 0 and 1 in the prior test, so guess it really isn't needed anyway especially as the test for 0 and 1 is testing not only count but also specific values so it is more detailed.